### ISSUE TITLE
body.Check is not a function. (In 'body.Check(c.body)', 'body.Check' is undefined) in Elysia 1.0.0

### ISSUE THREAD

**Issue by tnfAngel** [elysiajs/elysia#542]
### What version of Elysia.JS is running?

1.0.0

### What platform is your computer?

Linux 5.15.146.1-microsoft-standard-WSL2 x86_64 x86_64

### What steps can reproduce the bug?

setting type to 'json' and using t.Object for body sends this response: {"name":"TypeError","message":"body.Check is not a function. (In 'body.Check(c.body)', 'body.Check' is undefined)"}

Repro:
```ts
import { Elysia, t } from 'elysia';

new Elysia()
    .post(
        '/test',
        () => ({
            test: 'true'
        }),
        {
            type: 'json',
            body: t.Object({
                hi: t.String()
            }),
        }
    )
    .listen(8080);
```

Reported on discord: https://discord.com/channels/1044804142461362206/1044807567022493766/1218896455641862284

### What is the expected behavior?

Do not see this error

### What do you see instead?

 {"name":"TypeError","message":"body.Check is not a function. (In 'body.Check(c.body)', 'body.Check' is undefined)"}

### Additional information

Nothing is logged into the console. Is just the HTTP POST response.

**Comment by kofta999** [comment]
It seems that the problem is because of name conflict in `body` variable, when I tried to log the handler function
```ts
const {
                handler,
                handleError,
                hooks: {
                        transform,
                        resolve,
                        beforeHandle,
                        afterHandle,
                        mapResponse: onMapResponse,
                        parse,
                        error: handleErrors,
                        onResponse
                },
                validator: {
                        body,
                        headers,
                        params,
                        query,
                        response,
                        cookie
                },
                utils: {
                        mapResponse,
                        mapCompactResponse,
                        mapEarlyResponse,
                        parseQuery,
                        isNotEmpty
                },
                error: {
                        NotFoundError,
                        ValidationError,
                        InternalServerError,
                        ParseError
                },
                schema,
                definitions,
                ERROR_CODE,
                getReporter,
                requestId,
                parseCookie,
                signCookie,
                decodeURIComponent,
                ELYSIA_RESPONSE
        } = hooks



        return async function handle(c) {





                c.headers = c.request.headers.toJSON()

c.cookie = await parseCookie(c.set, c.headers.cookie, undefined)
if(c.qi !== -1) {
                                c.query = parseQuery(decodeURIComponent(c.request.url.slice(c.qi + 1)).replace(/\+/g, ' '))
                        } else c.query = {}
 try {
const body = await c.request.text()

                                                        try {
                                                                c.body = JSON.parse(body)
                                                        } catch {
                                                                throw new ParseError('Failed to parse body as found: ' + (typeof body === "string" ? "'" + body + "'" : body), body)
                                                        }

if(body.Check(c.body) === false) {
                        c.set.status = 422; throw new ValidationError('body', body, c.body)
                }let r = handler(c);
return mapResponse(r, c.set, c.request)

} catch(error) {const set = c.set
if (!set.status || set.status < 300) set.status = error?.status || 500

                                c.error = error
                                c.code = error.code ?? error[ERROR_CODE] ?? "UNKNOWN"

let er0 = handleErrors[0](c)
er0 = mapEarlyResponse(er0, set, c.request)
if (er0) {return er0 }
return handleError(c, error, true)

}
        }
```
It seems that the line
```ts
if (body.Check(c.body) === false)
```
uses the body defined in the `try` scope which is a string instead of the validator object defined at the beginning of the function
So I guess this should be fixed by just renaming the inner `body` variable in `try` block to `RequestBody` for example

I'll try doing it and submit a PR myself

**Comment by ethndotsh** [comment]
I am experiencing this error as well.

**Comment by asphum** [comment]
delete type: "json" and try agian.

**Comment by SaltyAom** [comment]
Should be fixed in e2d2620, published on 1.0.1

**Comment by tnfAngel** [comment]
Fixed!
